File: THCS/2023_2024/CanTho_2324/Bai1/QUA.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    string input_path = "QUA.INP";
    string output_path = "QUA.OUT";

    // Mở tệp nhập
    ifstream infile(input_path);
    if (!infile.is_open()) {
        cerr << "Không thể mở tệp: " << input_path << endl;
        return 1;
    }

    int n, k;
    infile >> n >> k;

    // Tạo vector để lưu giá trị quà
    vector<int> values(n);

    // Đọc giá trị của từng phần quà
    for (int i = 0; i < n; ++i) {
        infile >> values[i];
    }

    infile.close();

    // Sắp xếp giá trị quà theo thứ tự giảm dần
    sort(values.begin(), values.end(), greater<int>());

    // Tính tổng giá trị lớn nhất của k phần quà
    int max_value = 0;
    for (int i = 0; i < k; ++i) {
        max_value += values[i];
    }

    // Mở tệp xuất
    ofstream outfile(output_path);
    if (!outfile.is_open()) {
        cerr << "Không thể mở tệp: " << output_path << endl;
        return 1;
    }

    outfile << max_value << endl;

    outfile.close();

    return 0;
}

File: THCS/2023_2024/CanTho_2324/Bai1/QUA.py
input_path = "./QUA.INP"
output_path = "./QUA.OUT"

def select_gifts(input_file, output_file):
    with open(input_file, 'r') as infile:
        # Đọc n và k
        n, k = map(int, infile.readline().strip().split())

        # Đọc giá trị của từng phần quà
        values = list(map(int, infile.readline().strip().split()))

    # Sắp xếp giá trị quà theo thứ tự giảm dần
    values.sort(reverse=True)

    # Tính tổng giá trị lớn nhất của k phần quà
    max_value = sum(values[:k])

    # Ghi kết quả vào tệp
    with open(output_file, 'w') as outfile:
        outfile.write(f"{max_value}\n")

select_gifts(input_path, output_path)
